src/components/implementation/no_interface/cbboot/quarantine.c

```diff
@@ -295,126 +302,136 @@ quarantine_fork(spdid_t spdid, spdid_t source)
 #endif
 
 	/* inform servers about fork. Have to let servers update
-	 * spdid-based metadata before either the source (orig) or d_spd (fork)
+	 * spdid-based metadata before either the source (orig) or dest (fork)
 	 * make invocations to the servers. This is also done lazily through
 	 * the upcall mechanism in the fault handling path. */
 	/* should iterate the forked spd's deps and inform each? */
 	/* mman: have to do this now so the memory maps are available. */
 	if (tot > SERVICE_SIZE) tot = SERVICE_SIZE + 3 * round_up_to_pgd_page(1) - tot;
 	else tot = SERVICE_SIZE - tot;
-	printd("Telling mman to fork(%d, %d, %d, %x, %d)\n", cos_spd_id(), source, d_spd, prev_map + PAGE_SIZE, tot);
+	printd("Telling mman to fork(%d, %d, %d, %x, %d)\n", cos_spd_id(), source, dest, prev_map + PAGE_SIZE, tot);
 	
-	r = mman_fork_spd(cos_spd_id(), source, d_spd, prev_map + PAGE_SIZE, tot);
+	r = mman_fork_spd(cos_spd_id(), source, dest, prev_map + PAGE_SIZE, tot);
 	if (r) printc("Error (%d) in mman_fork_spd\n", r);
 
-#if 0
+#ifdef NIL
 	/* cbuf */
-	printd("Telling cbuf to fork(%d, %d, %d)\n", cos_spd_id(), source, d_spd);
-	r = cbuf_fork_spd(cos_spd_id(), source, d_spd);
+	printd("Telling cbuf to fork(%d, %d, %d)\n", cos_spd_id(), source, dest);
+	r = cbuf_fork_spd(cos_spd_id(), source, dest);
 	if (r) printc("Error (%d) in cbuf_fork_spd\n", r);
 
 	/* TODO: valloc */
 #endif
 
-	quarantine_add_to_spd_map(source, d_spd);
+	quarantine_add_to_spd_map(source, dest);
 
 #if QUARANTINE_MIGRATE_THREAD
-	quarantine_migrate(cos_spd_id(), source, d_spd, d_thd);
+	quarantine_migrate(cos_spd_id(), source, dest, d_thd);
 	printd("Waking up thread %d\n", d_thd);
 	if (d_thd) {
 		sched_quarantine_wakeup(cos_spd_id(), d_thd);
 	}
 #endif
 
 	/* TODO: should creation of boot threads be controlled by policy? */
-	printd("Creating boot threads in fork: %d\n", d_spd);
-	__boot_spd_thd(d_spd);
+	printd("Creating boot threads in fork: %d\n", dest);
+	__boot_spd_thd(dest);
+	int reset;
+	goto done;
+
+error:
+	do { printc("Found an error\n"); } while (0);
+	dest = 0;
 
 done:
-	assert(0 == sched_curr_set_priority(p));
-	printd("Forked %d -> %d\n", source, d_spd);
-	return d_spd;
+	reset = sched_curr_set_priority(p);
+	assert(0 == reset);
+	printd("Forked %d -> %d\n", source, dest);
+	return dest;
 }
 
-
+/*
+ * Handles routing of calls???
+ */
 int
 fault_quarantine_handler(spdid_t spdid, long cspd_dspd, int cap_ccnt_dcnt, void *ip)
 {
 	unsigned long r_ip;
 	int tid = cos_get_thd_id();
 	u16_t capid;
 	s8_t c_fix, d_fix;
-	int c_spd, d_spd;
+	int c_spd, dest;
 	int f_spd;
 	int inc_val;
 
+	// unpack
 	capid = cap_ccnt_dcnt>>16;
-	d_fix = (cap_ccnt_dcnt>>8)&0xff; /* fix the d (server) if snd != 0 */
-	c_fix = cap_ccnt_dcnt&0xff; /* fix the c (client) if rcv != 0 */
+	d_fix = (cap_ccnt_dcnt>>8)&0xff; 	/* fix the d (server) if snd != 0 */
```

Comment:
style.  (I know that this code is from previous commits, but if you are fixing the alignment of comments, might as well fix the spacing around `&`)